data_handler.py

```diff
@@ -140,12 +151,46 @@ def main(self, datalog, cpm, cpm_err, this_start, this_end, counts):
         self.manager.data_log(datalog, cpm, cpm_err)
 
         if self.manager.test:
-            self.test_send(cpm, cpm_err)
+            self.no_network_send(cpm, cpm_err)
         elif not self.manager.config:
             self.no_config_send(cpm, cpm_err)
         elif not self.manager.publickey:
             self.no_publickey_send(cpm, cpm_err)
-        elif not self.manager.network_up:
-            self.no_network_send(cpm, cpm_err)
         else:
-            self.regular_send(this_end, cpm, cpm_err)
+            try:
+                self.regular_send(this_end, cpm, cpm_err)
+            except socket.gaierror as e:

```

Comment:
All of these `socket` errors check `e[0]` anyway. Why don't you catch them all:
```
except (socket.error, socket.gaierror, socket.timeout) as e:
```
and put all the `if e[0] ==` statements into one if/elif block.